head.go

```diff
@@ -1101,6 +1110,17 @@ func (h *Head) chunksRange(mint, maxt int64) *headChunkReader {
 	return &headChunkReader{head: h, mint: mint, maxt: maxt}
 }
 
+// Meta returns meta information about the head.
+func (h *Head) Meta() BlockMeta {
+	var id [16]byte
+	copy(id[:], "______head______")
```

Comment:
> If the id is going to remain same, why not have it as a global const? Would avoid allocation and copying every time it is called.

just double checked and can't see how to save a string/byte array into a const.


>PS: I am not sure what Meta for Head should actually mean as only MinTime and MaxTime are relevant here. I would be careful about adding Meta() to BlockReader, for example if I try to use BlockStats from the Meta() somewhere and if it happens to be Head, I would get wrong numbers.

yes I was thinking the same, but really want to use the BlockReader interface instead of blocks directly to make sure it doesn't provide write operations and the `analyzeBlock` func requires the meta do display the block UUID. 
I think if we run into some sort of problems can refactor later.

Reply:
> just checked and can't see how to save a string/byte array into a const.

I think that's fine then.

> I think if we run into some sort of problems can refactor later.

Hope we don't run into any problem!

Reply:
> PS: I am not sure what Meta for Head should actually mean as only MinTime and MaxTime are relevant here. I would be careful about adding Meta() to BlockReader, for example if I try to use BlockStats from the Meta() somewhere and if it happens to be Head, I would get wrong numbers.

Yea this is bit weird. Maybe splitting `Meta()` to actually `BlockStats` which would be current snapshot of things might be nicer? not sure, worth to comment on `Meta` interface method for sure. Head is already wrapped with `BlockReader` even though it is not a block so it's fine I think.

Reply:
Also please move this string to constant at least `const headBlockID = "______head______"` I think it might nice to keep in single place. Although we don't depend on it now so we can refactor later.

Reply:
resolved without addressing my questions.

Reply:
>resolved without addressing my questions.

not sure what happened, wasn't my intention to leave unanswered.

> Yea this is bit weird. Maybe splitting Meta() to actually BlockStats which would be current snapshot of things might be nicer?

that is a good idea, but how can we do it without braking users and the current meta json implementation?

> not sure, worth to comment on Meta interface method for sure. 

What would this comment say and why/how would it be useful?



>Also please move this string to constant at least const headBlockID = "______head______" I think it might nice to keep in single place. Although we don't depend on it now so we can refactor later.

Why not do it when/if we actually need to use it in more than a single place?

Reply:
> Why not do it when/if we actually need to use it in more than a single place?

Sure, just here this is quite vague ID. Putting in constant would allow to name it `headBlockID` so it would maybe more obvious (not a blocker)

> that is a good idea, but how can we do it without braking users and the current meta json implementation?

I think we misunderstood -> I meant rather having `MinTime` and `MaxTime` and `BlockStats` as separate methods. Anyway, same problem, so ignore me.

> What would this comment say and why/how would it be useful?
> > not sure, worth to comment on Meta interface method for sure.

That for head block this meta can give "dynamic" result for both max time and stats.

Reply:
updated.

Reply:
What exactly? (: